python/ray/ml/utils/remote_storage.py

```diff
@@ -100,7 +113,12 @@ def get_fs_and_path(
         # Raised when protocol not known
         return None, None
 
-    fs = pyarrow.fs.PyFileSystem(pyarrow.fs.FSSpecHandler(fsspec_fs))
+    fsspec_handler = pyarrow.fs.FSSpecHandler
+    if parsed.scheme in ["gs", "gcs"]:
+        # GS doesn't support `create_parents` arg in `create_dir()`
+        fsspec_handler = _CustomFSSpecHandler
```

Comment:
do we want to raise exception if _CustomFSSpecHandler is None here?
actually, what utilities does this offer if pyarrow.fs is missing? should we just raise, and simplify all the logics here?

Reply:
`pyarrow.fs` only supports `file://`, `hdfs://`, `s3://` and `mock://`. For other cloud providers (e.g. gs or azure) the user can install a fsspec-compatible extension (e.g. `gcsfs` for gs).

Reply:
At this path `_CustomFSSpecHandler` is not None, otherwise we would have returned in line 105

Reply:
I see. yeah, as Amog said, add a TODO to clean this up would be nice.